design/one-pager-consuming-k8s-clusters.md

```diff
@@ -0,0 +1,270 @@
+# Consuming Kubernetes Clusters
+* Owner: Dan Mangum (@hasheddan)
+* Reviewers: Crossplane Maintainers
+* Status: Draft
+
+## Terminology
+
+* Kubernetes `Provider`: a cluster-scoped custom resource that points to any
+  Kubernetes `Secret` that contains connection information for a Kubernetes
+  cluster.
+* Kubernetes cluster **managed resource**: a cluster-scoped custom resource that
+  represents the existence and configuration of a specific cloud provider's
+  managed Kubernetes service (e.g. GKE, AKS, EKS).
+* `KubernetesCluster` claim: a namespace-scoped custom resource that can be used
+  to claim a Kubernetes cluster managed resource, or dynamically provision one.
+* **Workload**: a collection of Kubernetes resources that are intended to be
+  deployed into a target Kubernetes cluster. A workload is represented by a
+  `KubernetesApplication` custom resource.
+
+## Background
+
+Currently, Crossplane models portable cloud provider managed services as
+*managed resources* that can be either statically or dynamically provisioned
+using resource *classes* and *claims*. One such resource type is a Kubernetes
+cluster, which uses a `KubernetesCluster` claim and cloud provider-specific
+managed resource types (e.g. `GKECluster`, `AKSCluster`, `EKSCluster`, etc.). It
+is appropriate for Kubernetes clusters to be modeled in the same manner as other
+managed services due to the fact that they are presented by the provider in the
+same fashion. However, in Crossplane, Kubernetes clusters are unique to other
+managed service offerings in that they can also be used as a target for running
+workloads.
+
+Workloads are provisioned using the `KubernetesApplication` type. It is
+namespace-scoped and is scheduled to a target cluster using labels to match a
+`KubernetesCluster` claim in its namespace. Because managed resources may only
+be bound to a single claim, the claim is namespaced, and `KubernetesApplication`
+resources can only be scheduled to `KubernetesCluster` claims in their
+namespace, any Kubernetes managed resource offering may only be consumed (i.e.
+deployed to) from a single namespace.
+
+In order to fully conceptualize the usage of Kubernetes clusters in Crossplane,
+it is helpful to separate the notions of a **Kubernetes cluster as a managed
+service** and a **Kubernetes cluster as a target for deploying workloads**. With
+this distinction made, we can assert that this proposal in no way modifies the
+managed service notion of a Kubernetes cluster, but instead formalizes this
+concept of a Kubernetes cluster as a deployment target.
+
+## Goals
+
+Because it is unlikely that Crossplane users will always want a 1-to-1
+namespace-to-cluster consumption model, it is desirable to support deploying
+workloads in any namespace to any Kubernetes cluster. However, it is also
```

Comment:
I know what you're getting at here because we've discussed it out of band, but I don't know if it would be obvious to uninitiated readers. I wonder if we could expand on the use case(s) a little more to make this clearer. I'm going to try do that in this comment:

We're assuming Crossplane is deployed as a standalone** "control cluster" tasked with:

* Managing cloud infrastructure such as databases, buckets, and caches.
* Managing Kubernetes clusters - a type of cloud infrastructure.
* Managing the applications deployed to said Kubernetes clusters.

Each Crossplane control cluster is divided up into namespaces. _Possibly_ each namespace belongs to some application team; for example the payments team will create all the payments related resource claims and workloads in the `payments` namespace. We expect that the infrastructure operator configuring Crossplane for these teams to use may want to:

* Require all teams to share one Kubernetes cluster that is managed by the infrastructure operator.
* Require all teams to share a small set of Kubernetes clusters that are managed by the infrastructure operator.
* Allow each team to provision one or more Kubernetes clusters to deploy their workloads to.

Currently we support only the last strategy, but we think the first two strategies are likely to be more popular.

** All of this is possible when Crossplane is deployed into an existing Kubernetes cluster too, but it seems like an uncommon use case to install Crossplane onto an existing Kubernetes cluster, then use Crossplane to create _other_ Kubernetes clusters, and to deploy workloads to them.

Reply:
Aha, I see you went into this in detail further down into the doc. I'll leave my comment for posterity, but I don't think we need to take any action.